### PR TITLE
piksi_leds: Invert link LED when network is availble.

### PR DESCRIPTION
**Pull request by gsmcmullin**
For Duro Cell Modem solution S1.

This doesn't actually poll the network, but reads from a file `/var/network_available`.  That file must contain a '0' or a '1' filled in by another process.

For task https://github.com/swift-nav/firmware_team_planning/issues/430

### CODE REVIEW

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2018010520-13-g33175be
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2018010520-13-g33175be/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2018010520-13-g33175be/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2018010520-13-g33175be

File: package/piksi_leds/src/manage_led.c [hunk 1]
```diff
@@ -214,6 +218,10 @@ static void handle_link(rgb_led_state_t *s) {
     on_off = false;
   }
 
+  if (network_available && (*network_available == '1')) {
+    on_off = !on_off;
```
Comment (silverjam) [inline package/piksi_leds/src/manage_led.c]: I don't think this implements the behavior correctly, we're supposed to blink if we have obs + network, doing this here will just flip back whatever on_off was changed to by the obs check.

File: package/piksi_leds/src/manage_led.c [hunk 2]
```diff
@@ -293,7 +301,16 @@ static void * manage_led_thread(void *arg) {
...
 }
 
+void sigbus_handler(int signum) {
+  /* Just don't crash if the file is empty */
+  lseek(network_available_fd, 0, SEEK_SET);
+  write(network_available_fd, "0", 1);
+}
+
 void manage_led_setup(bool is_duro) {
+  network_available_fd = open("/var/network_available", O_CREAT | O_RDWR, 0644);
```
Comment (silverjam) [inline package/piksi_leds/src/manage_led.c]: Let's make this /var/run/network_available to be consistent with nmea_daemon.

File: package/piksi_leds/src/manage_led.c (hunk 1 above)
Comment (gsmcmullin) [reply package/piksi_leds/src/manage_led.c]: This inverts it.  obs with no network is normally off with short blinks on, obs with network is normally on with short blinks off.

Comment (gsmcmullin) [reply package/piksi_leds/src/manage_led.c]: From the design doc:
> If red LED is solid on and blinks off intermittently, Duro is connected to the internet and is receiving observations from Skylark / a CORS"

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2018010520-19-gb7a6008
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2018010520-19-gb7a6008/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2018010520-19-gb7a6008/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2018010520-19-gb7a6008

**Pull request merged**